packages/react-swipeable-views-utils/src/bindKeyboard.js

```diff
@@ -37,10 +37,11 @@ export default function bindKeyboard(MyComponent) {
       });
     }
 
-    componentWillReceiveProps(nextProps) {
```

Comment:
I think you should use getDerivedStateFromProps instead, 

something like

static getDerivedStateFromProps(props, state) {
      const { index } = props;
      if (typeof index === 'number' && index !== state.lastIndex) {
        return {
          index: index,
          lastIndex: index
        };
      }
      return null;
    }